wagtail/wagtailcore/tests/test_blocks.py

```diff
@@ -76,6 +85,17 @@ def test_charfield_render_with_template_with_extra_context(self):
 
         self.assertEqual(html, '<h1 lang="fr">Bonjour le monde!</h1>')
 
+    def test_charfield_get_context_with_context(self):
+        block = blocks.CharBlock(template='tests/blocks/heading_block.html')
+        context = {
+            'language': 'fr',
+        }
+        ret_context = block.get_context('hello', context.copy())
+
+        self.assertEqual(context['language'], ret_context['language'])
+        self.assertEqual('hello', ret_context['self'])
+        self.assertEqual('hello', ret_context['value'])
+
```

Comment:
One thing we're missing here is an end-to-end test to confirm that context passed to `render` is successfully passed on to `get_context`. Perhaps the best way to do this is to update https://github.com/wagtail/wagtail/blob/0e9c255b034ce4bdef5b265021833790fd813ed9/wagtail/tests/testapp/blocks.py#L10 to do something meaningful with the parent context, and extend the test here: https://github.com/wagtail/wagtail/blob/0e9c255b034ce4bdef5b265021833790fd813ed9/wagtail/wagtailcore/tests/test_blocks.py#L2361